Objects/typeobject.c

```diff
@@ -6229,6 +6229,10 @@ PyType_Ready(PyTypeObject *type)
 
     type->tp_flags |= Py_TPFLAGS_READYING;
 
+    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE)) {
```

Comment:
Please add a short comment to explain that with a reference to bpo-43908. I understood that for historical reasons, all static types are automatically immutable.

Reply:
Good point. Thanks, I'll do that. Also, thanks for the doc improvements!

Reply:
I suggest simply:

```
// bpo-43908: Static types are always immutable.
```

Reply:
See b7560e44ade6312656a3db94f1d8889caeb341c7

Reply:
That comment just doubles what the code says. I would keep it a little bit more informative.